### DISCUSSION TITLE
Plan start time for each demand in forward scheduling?

### DISCUSSION THREAD

**Discussion by binghui1992** [frePPLe/frepple#586] [Q&A] [answered]
I wonder if I could setup a plan start time for each demand in forward sheduling in cloud edition or even in community edition, which means the planned operation can not be started sooner than this time.

Well, I did not see any options related to backward or forward scheduling in the "execution" page but on the Plan Editor  (I'm using the cloud edition). In addition, I do see the forward scheduling button in the Plan Editor page, however I did not see options for setting up a plan start time for all demands or even for each demands (the default time is from the current date, right?).

BTW I've read through some references and forums but no found much intrests: https://frepple.com/blog/backward-or-forward-scheduling/

**Comment by jdetaeye** [comment, marked as answer]
There is a global parameter "plan.administrativeLeadtime" that is used to control the backward or forward scheduling mode.  It specifies a time we try to have a sales order order ready before its due date.

The default is 0 (days), meaning just-in-time mode, backward from the due date.

Setting the paramerter to a high value like 999 (days) effectively makes the solver work in ASAP forward planning mode.

>>I wonder if I could setup a plan start time for each demand in forward sheduling in cloud edition or even in community edition, which means the planned operation can not be started sooner than this time.

The parameter described above is a global setting.  
We have implemented some customization for a customer which is controlling a "don't-ship-before" date per demand.  (Behind the scenes the customization is dynamically updating on the fly the plan.administrativeLeadtime parameter in the solver)

**Reply by binghui1992** [reply]
Thanks for the reply, I've not dive much into the source code since I'm a junior devoloper with little experience on C++, will look deeper into it..

**Reply by jdetaeye** [reply]
This level of customization is possible through python code.   The C++ solver code itself is generic for all our customers. Having compiled-per-customer solvers wouldn't be supportable for us...
